torch/_inductor/ir.py

```diff
@@ -407,16 +408,31 @@ def get_reads(self):
         with patch.object(FlexibleLayout, "allow_indexing", True):
             if self.get_reduction_type():
                 return extract_read_writes(
-                    self.make_loader(),
+                    self.make_loader(),  # type: ignore[attr-defined]
```

Comment:
Can we fix this one (and most of the other attr-defined's by adding an abstract method to the base class?

Reply:
That isn't exactly semantics-preserving because of point 2 -- defining it on the base class (IRNode) means that MutableBox's `__getattr__` won't end up looking the method up on its `self.data` member. I did try defining one of these methods and tests did indeed fail. I do agree that this isn't very clean though...

Reply:
One option would be to do away with `MutableBox.__getattr__` altogether, and require every method to be explicitly defined instead. That would make evolving the IR less flexible, not sure if that's a concern at all

But we do probably need something like this at some point if we wish to enable `follow_imports` when typechecking `_inductor/` -- there would be way too many errors otherwise.

Reply:
IMO MutableBox shouldn't have been implemented with a `__getattr__` but this is a hot take

Reply:
Agreed, you can just add an explicit "foward to .data" method in MutableBox.